src/components/create-safe/status/usePendingSafeCreation.ts

```diff
@@ -24,11 +25,24 @@ export const pollSafeInfo = async (chainId: string, safeAddress: string): Promis
   })
 }
 
-export const checkSafeCreationTx = async (provider: JsonRpcProvider, txHash: string, chainId: string) => {
-  const TIMEOUT_TIME = 6.5
+export const checkSafeCreationTx = async (
+  provider: JsonRpcProvider,
+  txHash: string,
+  chainId: string,
+): Promise<SafeCreationStatus> => {
+  const TIMEOUT_TIME = 6.5 * 60 * 1000 // 6.5 minutes
 
   try {
     const txResponse = await provider.getTransaction(txHash)
+
+    if (!txResponse) {
```

Comment:
It was previously throwing because `txResponse` was undefined, and it assumed the Safe creation had failed.

Now it will return a timeout instead.

Reply:
makes sense